spring-boot-autoconfigure/src/main/java/org/axonframework/boot/autoconfig/AxonAutoConfiguration.java

```diff
@@ -86,58 +97,68 @@ public AxonAutoConfiguration(EventProcessorProperties eventProcessorProperties,
     }
 
     @Bean
-    @Primary
-    @ConditionalOnMissingQualifiedBean(beanClass = Serializer.class, qualifier = "!eventSerializer,messageSerializer")
-    public Serializer serializer(RevisionResolver revisionResolver) {
-        return buildSerializer(revisionResolver, serializerProperties.getGeneral());
+    @ConditionalOnMissingBean
+    public RevisionResolver revisionResolver() {
+        return new AnnotationRevisionResolver();
     }
 
-    private Serializer buildSerializer(RevisionResolver revisionResolver, SerializerProperties.SerializerType serializerType) {
-        switch (serializerType) {
-            case JACKSON:
-                return new JacksonSerializer(revisionResolver, new ChainingConverter(beanClassLoader));
-            case JAVA:
-                return new JavaSerializer(revisionResolver);
-            case XSTREAM:
-            case DEFAULT:
-            default:
-                XStreamSerializer xStreamSerializer = new XStreamSerializer(revisionResolver);
-                xStreamSerializer.getXStream().setClassLoader(beanClassLoader);
-                return xStreamSerializer;
+    @Bean
+    @ConditionalOnMissingBean
+    public ObjectMapper objectMapper() {
+        return new ObjectMapper();
```

Comment:
I would instantiate `ObjectMapper` like this: `new ObjectMapper().findAndRegisterModules();`. This way we would have all Jackson modules registered. Do you have anything against it?

Reply:
I like it, wasn't aware of that function.
Will adjust as suggested.